designer/client/inline-conditions.js

```diff
@@ -0,0 +1,200 @@
+
+    this.setState({
+      conditions: conditions.add(new Condition(Field.from(condition.field), condition.operator, Value.from(condition.value), condition.coordinator)),
+      condition: undefined
+    })
+  }
+
+  conditionFieldId (id, fieldName) {
+    return `conditions.${id}.${fieldName}`
+  }
+
+  parseConditionId (fieldId) {
+    return fieldId.split('.')[1]
+  }
+
+  onChangeField = e => {
+    const input = e.target
+    const fieldName = input.value
+
+    const { condition } = this.state
+
+    const currentField = condition.field ? condition.field.name : undefined
+    const currentOperator = condition.operator
+
+    this._updateCondition(condition, c => {
+      if (fieldName) {
+        if (currentField && this.fields[currentField].values !== this.fields[fieldName].values) {
+          delete c.value
+        }
+        if (currentOperator && !conditionalOperators.getConditionals(fieldName)[currentOperator]) {
+          delete c.operator
+        }
+        c.field = new Field(fieldName, this.fields[fieldName].label)
+      } else {
+        delete c.field
+        delete c.operator
+        delete c.value
+      }
+    })
+  }
+
+  _updateCondition (condition, updates) {
+    const copy = clone(condition)
+    updates(copy)
+    this.setState({
+      condition: copy
+    })
+  }
+
+  onChangeOperator = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.operator = input.value })
+  }
+
+  onChangeValue = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.value = new Value(input.value) })
+  }
+
+  render () {
+    const { conditions, condition } = this.state
+
+    const fieldDef = condition && condition.field ? this.fields[condition.field.name] : undefined
+
+    const hasConditions = conditions.hasConditions()
+
+    return (
+      <div>
+        { hasConditions &&
+          <div className='govuk-form-group'>
+            <label className='govuk-label govuk-label--s' htmlFor='condition-string'>When</label>
+            <div key='condition-string' >
+              {conditions.toPresentationString()}
+            </div>
+          </div>
+        }
+        { condition &&
+        <div className='govuk-form-group'>
+          <label className='govuk-label govuk-label--s' htmlFor='field'>{hasConditions ? condition.coordinator : 'When'}</label>
+          <select className='govuk-select' id='cond-field' name='cond-field' value={condition && condition.field ? condition.field.name : undefined}
+            onChange={this.onChangeField}>
+            <option />
+            {Object.values(this.fields).map(field => {
+              return <option key={field.name} value={field.name}>{field.label}</option>
+            })}
+          </select>
+
+          { fieldDef &&
+          <select className='govuk-select' id='cond-operator' name='cond-operator' value={condition.operator || undefined}
```

Comment:
I don't think you need to do `|| undefined`. it will just be undefined if it is so

Reply:
Yeah, I'm not sure what I was thinking there!